src/core/wrap.ts

```diff
@@ -0,0 +1,18 @@
+import { Metric } from "@clarity-types/data";
+import config from "@src/core/config";
+import * as metric from "@src/data/metric";
+
+// tslint:disable-next-line: ban-types
+export default function(method: Function): Function {
+    return function(): void {
+        let start = Date.now();
+        method.apply(this, arguments);
+        let duration = Date.now() - start;
```

Comment:
Why not use perforamnce.now() for timing? More precise.

Reply:
Good point. Using performance.now() now and rounding off at the time of encoding so we don't have to deal with floating points while uploading.